### ISSUE TITLE
Compile error with clang++ 4.0.0 vs. DirTree

### ISSUE THREAD

**Issue by gvansickle** [gvansickle/ucg#118]
clang++ 4.0.0 doesn't like the *_basename_filter_type's in DirTree.h:

```
In file included from ../../../src/libext/DirTree.cpp:22:
../../../src/libext/DirTree.h:143:28: error: implicit instantiation of undefined template 'std::function<bool
      (const std::__cxx11::basic_string<char> &) noexcept>'
        file_basename_filter_type m_file_basename_filter;
                                  ^
```

**Comment by gvansickle** [comment]
Ends up it's the `noexcept` in the std::function<> template param.

**Comment by gvansickle** [comment]
Resolved in 395f89046b99afa2807b8aa6193c2ce103535431.

**Comment by fxcoudert** [comment]
This compilation error prevents us from offering ucg to Sierra & High Sierra (macOS 10.12 & 10.13) users in Homebrew. Could you consider making a minor release with this bug fix? Thanks
